msteams-platform/messaging-extensions/how-to/authentication/msgex-sso-graph-api.md

```diff
@@ -0,0 +1,144 @@
+
+> [!NOTE]
+>
+> - If your bot app hasn't been granted IT admin consent, app users have to provide consent the first time they use your app on a different platform.
+> - Implicit grant is not required if SSO is enabled on a bot app.
+
+You can configure authentication for multiple platforms as long as the URL is unique.
+
+### To configure authentication for a platform
+
+1. Open the app you registered in the [Azure portal](https://ms.portal.azure.com/).
+
+1. Select **Manage** > **Authentication** from the left pane.
+
+    :::image type="content" source="../../../assets/images/authentication/teams-sso-bots/azure-portal-platform.png" alt-text="Authenticate for platforms" border="true":::
+
+    The **Platform configurations** page appears.
+
+1. Select **+ Add a platform**.
+
+    :::image type="content" source="../../../assets/images/authentication/teams-sso-bots/add-platform.png" alt-text="Add a platforms" border="true":::
+
+    The **Configure platforms** page appears.
+
+1. Select the platform that you want to configure for your bot app. You can choose the platform type from web or SPA.
+
+    :::image type="content" source="../../../assets/images/authentication/teams-sso-bots/configure-platform.png" alt-text="Select web platform" border="true":::
+
+    You can configure multiple platforms for a particular platform type. Ensure that the redirect URI is unique for every platform you configure.
+
+    The **Configure Web** page appears.
+
+    > [!NOTE]
+    > The configurations will be different based on the platform you select.
+
+1. Enter the configuration details for the platform.
+
+    :::image type="content" source="../../../assets/images/authentication/teams-sso-bots/config-web-platform.png" alt-text="Configure web platform" border="true":::
+
+    1. Enter the redirect URI. The URI should be unique.
+    2. Enter the front-channel log-out URL.
+    3. Select the tokens you want Azure AD to send for your app.
+
+1. Select **Configure**.
+
+    The platform is configured and displayed in the **Platform configurations** page.
+
+## Acquire access token for MS Graph
+
+You'll need to acquire access token for Microsoft Graph.
+
+### Configure code to fetch access token
+
+The following code provides an example of using the access token obtained in the OAuth process to submit a request for permissions to Microsoft Graph.
+
+```TypeScript
+    public async displayMicrosoftGraphDataStep(stepContext: WaterfallStepContext): Promise<DialogTurnResult> {
+      // get token from prev step (or directly from the prompt itself)
+      const tokenResponse = stepContext.result;
+      if (!tokenResponse?.token) {
+        await stepContext.context.sendActivity("Login not successful, please try again.");
+      } else {
+        const msGraphClient = new MsGraphHelper(tokenResponse?.token);
```

Comment:
The `MsGraphHelper` is defined in the sample not in the SDK. It would be wise to either include its definition in the snippet _or_ to just directly create the graph API client.  

If a developer tries to drop this into their code this will fail to transpile.

Reply:
Requested for a relevant code sample for Graph permissions.

Reply:
Code updated